src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -140,6 +152,119 @@ private List<PluginFile> GetPluginFiles(CancellationToken cancellationToken)
             return files;
         }
 
+        /// <summary>
+        /// Gets auth plugins installed using dotnet tools. This is done by iterating through each file in directories found int eh
+        /// `PATH` environment variable.
+        /// The files are also expected to have a name that starts with `nuget-plugin-`
+        /// </summary>
+        /// <returns></returns>
+        private List<PluginFile> GetNetToolsPluginFiles()
+        {
+            var pluginFiles = new List<PluginFile>();
+
+            string[] paths = Array.Empty<string>();
+
+            // The path to the plugins installed using dotnet tools, should be specified in the NUGET_PLUGIN_PATHS environment variable.
+            paths = _environmentVariableReader.GetEnvironmentVariable("NUGET_PLUGIN_PATHS")?.Split(Path.PathSeparator) ?? Array.Empty<string>();
```

Comment:
If the `NUGET_PLUGIN_PATHS` environment variable is already set, the `_rawpluginpaths` parameter passed to the constructor will contain a list of paths to plugin executables that are installed as .NET tools going forward.

https://github.com/NuGet/NuGet.Client/blob/fef6562e0ebdc57c904a08f732cc917fdb4d555e/src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs#L320-L323

I think we should remove the logic in `PluginManager` for discovering .NET tool plugins because they have different verification logic. For example, the file name should start with `nuget-plugin-*`, the executable bit should be set for non-Windows platforms, and the file should have a .exe or .bat extension on Windows.

Reply:
I removed it and now the logic is
- If `NUGET_NETFX_PLUGIN_PATHS` or  `NUGET_NETCORE_PLUGIN_PATHS` are set, use them. 
- Otherwise if  `NUGET_PLUGIN_PATHS` is set, assume it is pointing to the directory that contains the dotnet tools installed plugins. If none of the env variables are set, go a head and scan for the plugins from the directories listed in PATH.

This limits `NUGET_PLUGIN_PATHS` to be used for .NET tools plugins only. Would that not be a problem for users who used for non .NET tools plugins previously?

Reply:
Great question, Nigusu. https://github.com/NuGet/Home/pull/13242#discussion_r1548145738 conversation in the spec, led us to the decision of using the existing environment variable for .NET tools plugins. There are few references to this environment variable in other repositories such as JetBrains, OneGet to name a few https://github.com/search?q=NUGET_PLUGIN_PATHS&type=code. The idea is to allow customers to install .NET tools as tool path tools instead of global tools. Maybe we can add a breaking change label to the issue so that it can be part of the release notes.

Reply:
I don't follow why we have to differentiate between dotnet tool and non dotnet tool with NUGET_PLUGIN_PATHS. 
Don't we just execute the with the full path of whatever was passed in there and how it starts it not a concern?

Reply:
On .NET (core) plugins using the old discovery method are executed via `dotnet {file}.dll`, whereas the new executables are executed directly, since they're executables.

Reply:
> Great question, Nigusu. [NuGet/Home#13242 (comment)](https://github.com/NuGet/Home/pull/13242#discussion_r1548145738) conversation in the spec, led us to the decision of using the existing environment variable (NUGET_PLUGIN_PATHS) for .NET tools plugins. There are few references to this environment variable in other repositories such as JetBrains, OneGet to name a few https://github.com/search?q=NUGET_PLUGIN_PATHS&type=code. The idea is to allow customers to install .NET tools as tool path tools instead of global tools. Maybe we can add a breaking change label to the issue so that it can be part of the release notes.

How about we support both instead of making it a breaking change. We could allow this environmental varibale to point to both .net tools plugins and plugin executables (.exe or .dll just like before). So the logic would be
* If the environment variable is point to an executable, treat it as a non .net tools plugin. This helps us avoid the breaking change
* If it is pointing to a folder, start searching for .net tools plugins in that folder.

Reply:
If it points to an executable, and that executable filename starts with `nuget-plugin-`, then you could treat it as a .net tool plugin as well.